posthog/api/feature_flag.py

```diff
@@ -56,6 +59,15 @@ def update(self, instance: FeatureFlag, validated_data: Dict, *args: Any, **kwar
         except IntegrityError:
             raise serializers.ValidationError("This key already exists.", code="key-exists")
 
+    def perform_destroy(self, key: str) -> bool:
+        featureFlag = FeatureFlag.objects.all().filter(deleted=True, key=key).first()
```

Comment:
This needs a filter by the current team: `.filter(team=team)` otherwise you might accidentally remove a key from another team

Reply:
Great, thanks for noticing.